util.py

```diff
@@ -5,25 +5,31 @@
 from constants import *
 
 
-def find_resume_json_weights(args):
-    enclosing_dir = os.path.join(args.model_save, args.resume_subdir)
+def find_resume_json_weights_args(args):
+    return find_resume_json_weights_str(args.model_save, args.resume_subdir, args.resume_epoch)
+
+def find_resume_json_weights_str(model_base_dir, model_subdir, resume_epoch):
+    enclosing_dir = os.path.join(model_base_dir, model_subdir)
 
     files = os.listdir(enclosing_dir)
 
     model_jsons         = [f for f in files if (".json" in f and HPE_HOURGLASS_STACKS_PREFIX in f)]
-    model_saved_weights = [f for f in files if (".hdf5" in f and "{prefix}{epoch:02d}".format(prefix=HPE_EPOCH_PREFIX, epoch=args.resume_epoch) in f)]
+    model_saved_weights = [f for f in files if (".hdf5" in f and "{prefix}{epoch:02d}".format(prefix=HPE_EPOCH_PREFIX, epoch=resume_epoch) in f)]
     
     assert len(model_jsons) > 0, "Subdirectory does not contain any model architecture json files"
     assert len(model_saved_weights) > 0, "Subdirectory does not contain any saved model weights"
 
-    args.resume_json = os.path.join(enclosing_dir, model_jsons[0])
-    args.resume_weights = os.path.join(enclosing_dir, model_saved_weights[0])
+    resume_json = os.path.join(enclosing_dir, model_jsons[0])
+    resume_weights = os.path.join(enclosing_dir, model_saved_weights[0])
+
+    print('Found model json: {}'.format(resume_json))
+    print('Found model weights for epoch {epoch:02d}: {weight_file_name}'.format(epoch=resume_epoch, weight_file_name=resume_weights))
+
+    assert os.path.exists(resume_json)
+    assert os.path.exists(resume_weights)
```

Comment:
just as a FYI, if my changes merge before yours, just `git checkout --ours util.py` when you rebase on main (take main's `util.py`. I made some changes that conflict with this, but won't affect any of your branch otherwise.